### PR TITLE
Add Json source generators and mark the library as trimmable

### PR DESCRIPTION
**Pull request by eduardobragaxz**
Also add .NET 8 as framework target since stable release is next week. I'm not entirely sure how that works, or if it was necessary. I did have to update the nuget packages to .NET 8 RC2 for the source generators to work, though.

~~Of note, I still see a trimming warning in TryDeserialize since I couldn't specify a type for the Deserialize method. I don't know how to fix that. Having said that, I haven't hit any errors on my app for now~~.

Fixed the last trim warning recently. Will continue testing it.

This is the first pull request I make ever, so I'm not sure if I'm doing it right.

### CODE REVIEW

**Comment by eduardobragaxz** [comment]
Somehow only three changes are unrelated, so I think it'd be better if you took a look at all of them. I'll stop making new commits 😄

**Comment by eduardobragaxz** [comment]
All of the properties in the History class are int, even though they come as string. I wrote a json converter to parse it to int, but I'm wondering if it would be better to just make them strings instead. It would technically be a breaking change, but getting trending hashtags already doesn't work in the current release.

File: Mastonet/MastodonClient.Status.cs [hunk 1]
```diff
@@ -147,7 +147,7 @@ public Task<MastodonList<Account>> GetFavouritedBy(string statusId, ArrayOptions
             data.Add(new KeyValuePair<string, string>("poll[expires_in]", poll.ExpiresIn.TotalSeconds.ToString()));
             if (poll.Multiple.HasValue)
             {
-                data.Add(new KeyValuePair<string, string>("poll[multiple]", poll.Multiple.Value.ToString()));
+                data.Add(new KeyValuePair<string, string>("poll[multiple]", poll.Multiple.Value.ToString().ToLowerInvariant()));
```
Comment (glacasa) [inline Mastonet/MastodonClient.Status.cs:150]: Did you notice a bug if we send True/False without lowering ?
If so, maybe we should add this `.ToLowerInvariant() ` also on line 155 for HideTotals

File: Mastonet/TimelineWebSocketStreaming.cs [hunk 2]
```diff
@@ -115,3 +121,8 @@ public override void Stop()
         base.Stop();
     }
 }
+
+[JsonSerializable(typeof(TimelineMessage))]
+internal partial class TimelineMessageContext : JsonSerializerContext
```
Comment (glacasa) [inline Mastonet/TimelineWebSocketStreaming.cs:126]: Can we put this class at the same level of `TimelineMessage`, so we can keep the visibility private ?

File: Mastonet.Entities/Enums/FilterContext.cs [hunk 3]
```diff
@@ -21,8 +22,13 @@ public class FilterContextConverter : JsonConverter<FilterContext>
     public override FilterContext Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
     {
         FilterContext context = 0;
-        var contextStrings = JsonSerializer.Deserialize<IEnumerable<string>>(ref reader, options);
-        if (contextStrings != null) {
+#if NET6_0_OR_GREATER
+        var contextStrings = JsonSerializer.Deserialize(ref reader, EntitiesContext.Default.IEnumerableString);
```
[the remark this answers is not in the document]
Comment (eduardobragaxz) [reply Mastonet.Entities/Enums/FilterContext.cs:26]: Do you mean it's not used when it targets .NET 5 and below? If so, maybe we can remove it. I kind of added it on automatic to match the one with EntitiesContext, but that might not be necessary.

[the remark this answers is not in the document]
Comment (glacasa) [reply Mastonet.Entities/Enums/FilterContext.cs:26]: No, it's the other way : 

On .NET Standard, it uses the old method :  
    JsonSerializer.Deserialize<IEnumerable<string>>(ref reader, options);

On .NET 6 or greater, the new method doesn't use `options` : 
    JsonSerializer.Deserialize(ref reader, EntitiesContext.Default.IEnumerableString);

File: Mastonet/MastodonClient.Status.cs (hunk 1 above)
Comment (eduardobragaxz) [reply Mastonet/MastodonClient.Status.cs:150]: I did. It defaults to false when not lowering. And yeah, happens to HideTotals too. I'm gonna add it.

File: Mastonet/TimelineWebSocketStreaming.cs (hunk 2 above)
Comment (eduardobragaxz) [reply Mastonet/TimelineWebSocketStreaming.cs:126]: I don't think so. If I add it right below TimelineMessage it throws three errors, and "fixing" (i.e. let Visual Studio fix) those throws an error in the Deserialize call that uses the context.

**Pull request merged**